subsys/shell/shell_ops.c

```diff
@@ -250,8 +250,29 @@ static void reprint_from_cursor(const struct shell *sh, uint16_t diff,
 			z_shell_raw_fprintf(sh->fprintf_ctx, "*");
 		}
 	} else {
-		z_shell_fprintf(sh, SHELL_NORMAL, "%s",
-			      &sh->ctx->cmd_buff[sh->ctx->cmd_buff_pos]);
+		/* Check if the reprint will cross a line boundary */
+		int line_len = sh->ctx->cmd_buff_len + z_shell_strlen(sh->ctx->prompt);
+		int buff_pos = sh->ctx->cmd_buff_pos + z_shell_strlen(sh->ctx->prompt);
+
+		if ((buff_pos / sh->ctx->vt100_ctx.cons.terminal_wid) !=
```

Comment:
I don't really like the requirement to perform division here to check if the reprint will cross a terminal width boundary, but I'm not sure how else to check this effectively- the condition we need to detect is one where the command buffer position is on a different line than the end of the command buffer

Reply:
I understand that you're wrapping each line manually. From my experience, most terminals automatically handle line wrapping when printing long commands. Could you share more details about the issue you're facing? What terminal and settings are you using?